flang/lib/Semantics/check-call.cpp

```diff
@@ -1600,6 +1600,18 @@ static void CheckMaxMin(const characteristics::Procedure &proc,
   }
 }
 
+static void CheckFree(evaluate::ActualArguments &arguments,
```

Comment:
Please could you add a similar check for the argument of MALLOC. I'm worried that we could have a compiler crash if somebody passes a type that cannot be converted to the runtime function argument type.

Reply:
I think we need this for basically all the intrinsics in the big type table in intrinsics.cpp. I reckon we should do that generally and generically for all of them, as a separate patch to this.